src/types.ts

```diff
@@ -104,9 +105,18 @@ export interface Context<T = any> {
    */
   contextId: number
 
+  /**
+   * Describes the language processor to be used for evaluation
+   */
   executionMethod: ExecutionMethod
 
   evaluationMethod: EvaluationMethod
+
+  /**
+   * Describes the strategy / paradigm to be used for evaluation
+   * Examples: lazy, concurrent or non-deterministic
+   */
+  variant?: Variant // This will replace evaluationMethod
```

Comment:
I have added a new field `Variant` that captures the evaluation strategy / paradigm.

Because our non-deterministic interpreter needs the value of this field when we create our prelude, I had to modify the `createContext` and `createEmptyContext` methods as well. This meant I also had to modify test code in other js-slang components which were using `createContext` and `createEmptyContext`.